### PR TITLE
chore(tree): a few refactors and instrumentations

### PR DESCRIPTION
**Pull request by mattsse**
just a few changes while getting more familiar with the implementation, mainly:

* add `CanonicalChain` wrapper type, this will make it easier to also store full blocks eventually
* split `try_insert_block` into multiple exclusive functions
* add more tracing

### CODE REVIEW

File: crates/blockchain-tree/src/block_indices.rs [hunk 1]
```diff
@@ -338,13 +338,58 @@ impl BlockIndices {
...
+    /// Canonical chain needed for execution of EVM. It should contains last 256 block hashes.
+    #[inline]
+    pub(crate) fn canonical_chain(&self) -> &CanonicalChain {
+        &self.canonical_chain
+    }
+}
+
+/// This keeps track of all blocks of the canonical chain.
+#[derive(Debug, Clone, Default)]
+pub(crate) struct CanonicalChain {
```
Comment (rakita) [inline crates/blockchain-tree/src/block_indices.rs:367]: nit: maybe can go in a separate file?

Comment (mattsse) [reply crates/blockchain-tree/src/block_indices.rs:367]: I like this here, because can keep everything private, because only written to by BlockIndices

**Review by rakita — APPROVED** [review]
one nit, but overall nice cleanup

File: crates/blockchain-tree/src/block_indices.rs (hunk 1 above)
Comment (mattsse) [reply crates/blockchain-tree/src/block_indices.rs:367]: but I think this will grow perhaps, if we add full blocks, so good point, will do

Comment (rakita) [reply crates/blockchain-tree/src/block_indices.rs:367]: It is not a big of deal, it can stay as it is, and if it grows we can always move it.

**Pull request merged**